sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs

```diff
@@ -368,6 +372,66 @@ public virtual async Task<Response<UploadBlobResult>> UploadBlobAsync(Stream str
             }
         }
 
+        private async Task<ChunkedUploadResult> UploadInChunks(string location, Stream stream, int chunkSize, bool async = true, CancellationToken cancellationToken = default)
+        {
+            // TODO: don't allocate a large buffer unless we need it.  How will we know?
+            byte[] buffer = new byte[chunkSize];
+            using SHA256 sha256 = SHA256.Create();
```

Comment:
I am just curious that what would happen if user passes chunkSize of 0? Or a really large number (100MB, 1GB) leading to OOM issues? Do we need to have upper and lower bounds for this value?

In case you missed it, see my previous comments here:
* https://github.com/Azure/azure-sdk-for-net/pull/32059#discussion_r1013369562
* https://github.com/Azure/azure-sdk-for-net/pull/32059#discussion_r1013368925

Reply:
Hi @ShivangiReja - sorry for my delay in responding! I was out sick for a few days, and I'm still a little slow catching up.

Yes, you are absolutely right.  If the caller passes [UploadBlobOptions](https://apiview.dev/Assemblies/Review/ce4598afc1f94854b136ea2183e05af6#Azure.Containers.ContainerRegistry.Specialized.UploadBlobOptions) with MaxChunkSize of 0, a zero-length byte array would be allocated, and the call would fail because all the stream.Read calls wouldn't have anywhere to copy the data from the stream to, and my while loop would be infinite.  Yikes!  I was planning to add argument validation to the UploadBlob method to prevent this case.

I wanted to do a little more research to see what the precedent was in other of our libraries like storage to understand whether we place limitation on the max size a caller can specify.  I'd personally be cautious about hard-coding something that makes an assumption about the memory limits on a user's system and might not age well.  An argument could be made that it's the user's responsibility to provide a reasonable limit if they want to override the default, and if they ask for more memory than they have, they have written a bug.  But if our team has discussed this and made an informed decision, I'm happy to follow precedent, I just need to figure out what that is.

Does that address your concern?

Reply:
>An argument could be made that it's the user's responsibility to provide a reasonable limit if they want to override the default, and if they ask for more memory than they have, they have written a bug

Why are we asking users to pass the maxChunkSize? Can we not set it internally? 

I see storage is no giving users any option to pass the max size. See Upload API [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs#L583) and [BlobUploadOptions](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Blobs/src/Models/BlobUploadOptions.cs).

Reply:
Huh!  Honestly?  I had seen it in the PartitionedUploader API and thought it was exposed internally.  Good call - I can fix this value in the client.  Thanks!

Reply:
Not sure if we are talking about the same thing, but this is in the [StorageTransferOptions](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Common/src/StorageTransferOptions.cs#L29).

Reply:
Thanks @JoshLove-msft, I did see it in a public API then.